python/lsst/ts/externalscripts/auxtel/latiss_base_align.py

```diff
@@ -0,0 +1,788 @@
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+            self.log.debug(f"Using target {self.cwfs_target} for cwfs.")
+        elif hasattr(config, "track_target"):
+            self.log.debug(f"Tracking target {config.track_target} for cwfs.")
+            self.cwfs_target = config.track_target.get("target_name", "cwfs_target")
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+            if "icrs" in config.track_target:
+                self.cwfs_target_ra = config.track_target["icrs"]["ra"]
+                self.cwfs_target_dec = config.track_target["icrs"]["dec"]
+        else:
+            self.log.debug("No target configured.")
+            self.cwfs_target = None
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+
+        self.filter = config.filter
+        self.grating = config.grating
+
+        # exposure time for the intra/extra images (in seconds)
+        self.exposure_time = config.exposure_time
+
+        self.acq_exposure_time = config.acq_exposure_time
+
+        # offset for the intra/extra images
+        self.dz = config.dz
+        # delta offset for extra focal image
+        self.extra_focal_offset = config.extra_focal_offset
+
+        self.large_defocus = config.large_defocus
+
+        self.threshold = config.threshold
+
+        self.coma_threshold = config.coma_threshold
+
+        self.offset_telescope = config.offset_telescope
+
+        self.max_iter = config.max_iter
+
+        self.reason = config.reason
+
+        self.program = config.program
+
+        self.take_detection_image = config.take_detection_image
+
+    def set_metadata(self, metadata: salobj.type_hints.BaseMsgType) -> None:
+        """Sets script metadata.
+
+        Parameters
+        ----------
+        metadata : `salobj.type_hints.BaseMsgType`
+            Script metadata topic. The information is set on the topic
+            directly.
+        """
+        # It takes about 300s to run the cwfs code, plus the two acquisition
+        # exposures.
+        metadata.duration = 300.0 + 2.0 * self.exposure_time
+        metadata.filter = f"{self.filter},{self.grating}"
+
+    async def arun(self, checkpoint: bool = False) -> None:
+        """Perform CWFS measurements and hexapod adjustments until the
+        thresholds are reached.
+
+        Parameters
+        ----------
+        checkpoint : `bool`, optional
+            Should issue checkpoints (default=False)?
+
+        Raises
+        ------
+        RuntimeError:
+            If coordinates are malformed.
+        """
+
+        if self.cwfs_target is not None and self.cwfs_target_dec is None:
+            if checkpoint:
+                await self.checkpoint(f"Slewing to object: {self.cwfs_target}")
+            await self.atcs.slew_object(
+                name=self.cwfs_target,
+                rot=0.0,
```

Comment:
Do we need to do this at rot=0 ? It adds overhead, especially when using the parallactic rotType.

Reply:
We don't need to do it at `rot=0.0` what do you suggest tough? should we add this as a parameter? or maybe try to get the current rotator position and use it in `PhysicalSky` mode?

Reply:
why not make the `rot` and `RotType` config parameters? Default is SkyAuto (so basically do whatever is fastest).

Feeding the current position to PhysicalSky seems like we're adding complication to simplify standard behavior?